packages/zowe-explorer-api/src/logger/IZoweLogger.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the *
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at *
+ * https://www.eclipse.org/legal/epl-v20.html                                      *
+ *                                                                                 *
+ * SPDX-License-Identifier: EPL-2.0                                                *
+ *                                                                                 *
+ * Copyright Contributors to the Zowe Project.                                     *
+ *                                                                                 *
+ */
+
+import * as loggerConfig from "../log4jsconfig.json";
+import * as path from "path";
+import { Logger } from "@zowe/imperative";
+
+export enum MessageSeverityEnum {
+    TRACE = 0,
+    DEBUG = 1,
+    INFO = 2,
+    WARN = 3,
+    ERROR = 4,
+    FATAL = 5,
+}
+
+/**
+ * Creates an instance of the Imperative logger for exxtenders to use
+ *
+ * @export
+ * @class IZoweLogger
+ */
+export class IZoweLogger {
+    private log: Logger;
+    private extensionName: string;
+    /**
+     * Creates an instance of the Imperative logger
+     *
+     */
+    public constructor(extensionName: string, loggingPath: string) {
+        for (const appenderName of Object.keys(loggerConfig.log4jsConfig.appenders)) {
+            loggerConfig.log4jsConfig.appenders[appenderName].filename = path.join(
+                loggingPath,
+                loggerConfig.log4jsConfig.appenders[appenderName].filename
+            );
+        }
+        Logger.initLogger(loggerConfig);
+        this.extensionName = extensionName;
+        this.log = Logger.getAppLogger();
+    }
+
+    public getExtensionName(): string {
+        return this.extensionName;
+    }
+
+    public getImperativeLogger(): Logger {
+        return this.log;
+    }
+
+    /**
+     * Log an error to the Imperative log
+     *
+     */
+    public logImperativeMessage(message: string, severity: MessageSeverityEnum): void {
+        const messageWithExtensionName = `Error in extension ${this.extensionName}: ${message}`;
+        switch (severity) {
+            case 0:
```

Comment:
Can you change the case into a more descriptive check like: 

case fatal: 
case trace:

You might manipulate some data but I think it would be helpful for extenders that it is a descriptive check rather than numbers